### PR TITLE
Async loading external automation steps

### PR DESCRIPTION
**Pull request by mike12345567**
## Description
This introduces the ability for the builder/server to load automations asynchronously in as they are needed. New automations can be added to the `automations` repo and they will automatically appear for users of the builder.

Essentially this pulls down a manifest from S3 at startup and combines it with the built in definitions, when the builder requests this it automatically uses the definition schema to allow the user to configure the various action inputs and the outputs are automatically added to the context.

It should be noted that most of the work of this was on the `automations` repo as it handles S3 bucket creation, all the manifest generation and the packages themselves, as well as bundling.

## Screenshots
Here is a video showing how easy it is to create a new automation for the system using the CLI that has been built into the `automations` repo.
https://www.loom.com/share/dd1a5cafb6b542eb9c220a1d9a3efb29

### CODE REVIEW

**Comment by aptkingston** [comment]
Just leaving this other loom here as a further example of the automation functionality - this time using Zapier to add records to a Google Sheet when an order is created.

https://www.loom.com/share/42b38f707f6a43959c7e480da6311135

As mentioned above, the [automations repo](https://github.com/Budibase/automations) handles the building and publication of automations, and further info can be found there.

File: packages/server/src/automations/automationUtils.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+const CouchDB = require("../db")
+
+module.exports.cleanMustache = string => {
```
Comment (shogunpurple) [inline packages/server/src/automations/automationUtils.js:3]: NAB: a little JSdoc to describe the logic of this function might be nice for the uninitiated

File: packages/server/src/automations/automationUtils.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          string.slice(baseIdx, baseIdx + idxChar) +
+          charToReplace[key] +
+          string.slice(baseIdx + idxChar + 1)
+      }
+    }
+  }
+  return string
+}
+
+module.exports.cleanInputValues = (inputs, schema) => {
```
Comment (shogunpurple) [inline packages/server/src/automations/automationUtils.js:28]: Same as above

**Review by shogunpurple — APPROVED** [review]
No blockers - nice work! LGTM. Something worth mentioning:

We have this in notion: https://www.notion.so/budibase/Developer-Guide-6210492a58a74181b615334320918026

Not a priority atm, but we should update this (even if we just copy + paste stuff from the automations repo) because it pertains to the old workflow setup.

**Pull request merged**